UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetByProductVersion.cs

```diff
@@ -0,0 +1,148 @@
+﻿using NUnit.Framework;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.FunctionalTests.Helper;
+using UKHO.ExchangeSetService.API.FunctionalTests.Models;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.FunctionalTests
+{
+    class ExchangeSetByProductVersion
+    {
+        private ExchangeSetApiClient ExchangesetApiClient { get; set; }
+        private TestConfiguration Config { get; set; }
+        public DataHelper Datahelper { get; set; }
+        public ProductVersionModel ProductVersionmodel { get; set; }
+      
+        [SetUp]
+        public void Setup()
+        {
+            Config = new TestConfiguration();
+            ExchangesetApiClient = new ExchangeSetApiClient(Config.EssBaseAddress);
+            ProductVersionmodel = new ProductVersionModel();
+            Datahelper = new DataHelper();
+           
+        }
+
+        [Test]
+        public async Task WhenICallTheApiWithAValidProductVersion_ThenASuccessStatusIsReturned()
+        {
+            List<ProductVersionModel> ProductVersiondata = new List<ProductVersionModel>(); 
+
+            ProductVersionmodel.ProductName = "AU895561";
+            ProductVersionmodel.EditionNumber = 4;
+            ProductVersionmodel.UpdateNumber = 1;
+                      
+            ProductVersiondata.Add(Datahelper.GetProductVersionModelData(ProductVersionmodel.ProductName, ProductVersionmodel.EditionNumber, ProductVersionmodel.UpdateNumber));
+          
+            ProductVersionmodel.ProductName = "GB100625";
+            ProductVersionmodel.EditionNumber = 6;
+            ProductVersionmodel.UpdateNumber = 0;
+
+            ProductVersiondata.Add(Datahelper.GetProductVersionModelData(ProductVersionmodel.ProductName, ProductVersionmodel.EditionNumber, ProductVersionmodel.UpdateNumber));
+
+            var apiresponse = await ExchangesetApiClient.GetProductVersionsAsync(ProductVersiondata);
+            Assert.AreEqual(200, (int)apiresponse.StatusCode, $"Exchange Set for Product version is  returned {apiresponse.StatusCode}, instead of theEditionNumber expected 200.");
```

Comment:
```suggestion
            Assert.AreEqual(200, (int)apiresponse.StatusCode, $"Incorrect status code {apiresponse.StatusCode} is returned, instead of expected 200.");
```

Reply:
I have done the changes as suggested.
Thanks Greg.